docs/internals.rst

```diff
@@ -1,552 +1,32 @@
 .. include:: global.rst.inc
-.. highlight:: none
 .. _internals:
 
 Internals
 =========
 
 .. toctree::
 
-    security
+    internals/security
+    internals/data-structures
 
 This page documents the internal data structures and storage
 mechanisms of |project_name|. It is partly based on `mailing list
 discussion about internals`_ and also on static code analysis.
 
+Borg uses a low-level, key-value store, the :ref:`repository`, and
+implements a more complex data structure on top of it, which is made
+up of the :ref:`manifest <manifest>`, :ref:`archives <archive>`,
+:ref:`items <item>` and data :ref:`chunks`.
 
-Repository and Archives
------------------------
+Each repository can hold multiple :ref:`archives <archive>`, which
+represent individual backups that contain a full archive of the files
+specified when the backup was performed.
 
-|project_name| stores its data in a `Repository`. Each repository can
-hold multiple `Archives`, which represent individual backups that
-contain a full archive of the files specified when the backup was
-performed. Deduplication is performed across multiple backups, both on
-data and metadata, using `Chunks` created by the chunker using the Buzhash_
+Deduplication is performed globally across all data in the repository
+(multiple backups and even multiple hosts), both on data and metadata,
+using :ref:`chunks` created by the chunker using the Buzhash_
```

Comment:
I guess that's technically correct. But also technically the manifest can't be deduplicated and archives aren't deduplicated since their object ID == archive ID.

I'm not sure whether this has to be encoded into this sentence?